renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1388,55 +1401,124 @@ void WrappedOpenGL::glCopyImageSubData(GLuint srcName, GLenum srcTarget, GLint s
                                  dstData.curType != eGL_TEXTURE_CUBE_MAP_ARRAY)
                                     ? RDCMAX(1, dstData.depth >> dstLevel)
                                     : dstData.depth;
-        if(srcX == 0 && srcY == 0 && srcZ == 0 && dstX == 0 && dstY == 0 && dstZ == 0 &&
-           srcLevelWidth == dstLevelWidth && srcLevelHeight == dstLevelHeight &&
-           srcLevelDepth == dstLevelDepth)
+        size_t dstSize = GetCompressedByteSize(dstLevelWidth, dstLevelHeight, dstLevelDepth,
+                                               dstData.internalFormat);
+
+        bytebuf *srcCdPtr = NULL;
+        bytebuf tempCd;
+        // if we have source compressed data to copy
+        if(srcData.compressedData.find(srcLevel) != srcData.compressedData.end())
         {
-          dstData.compressedData[dstLevel] = srcData.compressedData[srcLevel];
+          srcCdPtr = &srcData.compressedData[srcLevel];
         }
-        else
+        else if(!srcIsCompressed)
         {
-          rdcfixedarray<uint32_t, 3> srcBlockSize = GetCompressedBlockSize(srcData.internalFormat);
-          rdcfixedarray<uint32_t, 3> dstBlockSize = GetCompressedBlockSize(dstData.internalFormat);
+          if(srcData.curType == eGL_TEXTURE_2D || srcData.curType == eGL_TEXTURE_2D_ARRAY)
+          {
+            // try reading back without existing compressedData
+            RDCASSERT(!srcIsCompressed);
+
+            tempCd.resize(srcSize);
+            srcCdPtr = &tempCd;
+
+            GLuint packbuf = 0;
+            GL.glGetIntegerv(eGL_PIXEL_PACK_BUFFER_BINDING, (GLint *)&packbuf);
+            GLint align = 1;
+            GL.glGetIntegerv(eGL_PACK_ALIGNMENT, &align);
```

Comment:
There is a helper `PixelPackState` which does this push/pop for you, as well as a helper `ResetPixelPackState`, since alignment is not the only pack value that could interfere here.